sql/src/main/java/io/crate/action/sql/Session.java

```diff
@@ -588,13 +589,17 @@ private static void emitRowCountsToResultReceivers(UUID jobId,
 
     @Nullable
     public List<? extends DataType> getOutputTypes(String portalName) {
+        List<DataType> result = null;
         Portal portal = getSafePortal(portalName);
         AnalyzedStatement analyzedStatement = portal.boundOrUnboundStatement();
-        if (analyzedStatement instanceof AnalyzedRelation) {
-            return Symbols.typeView(((AnalyzedRelation) analyzedStatement).fields());
-        } else {
-            return null;
+        if (analyzedStatement instanceof ResultSetStatement) {
+            ResultSetStatement update = (ResultSetStatement) analyzedStatement;
```

Comment:
```suggestion
            ResultSetStatement statement = (ResultSetStatement) analyzedStatement;
```

`update` in this context is too specific, this can be any kind of statement.

Personally I'd also remove the local `result` variable and return directly (so you don't have to backtrack to see what has been assigned to `result`), but that's a personal preference so feel free to ignore.